src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -831,64 +840,152 @@ window.qBittorrent.DynamicTable ??= (() => {
-                    // more info: https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event
-                    tr.tabIndex = -1;
+            }
 
-                    const rowId = rows[rowPos]["rowId"];
-                    tr.setAttribute("data-row-id", rowId);
-                    tr["rowId"] = rowId;
-
-                    for (let k = 0; k < this.columns.length; ++k) {
-                        const td = document.createElement("td");
-                        if ((this.columns[k].visible === "0") || this.columns[k].force_hide)
-                            td.classList.add("invisible");
-                        tr.append(td);
-                    }
+            // set the scrollable height
+            this.table.style.height = `${rows.length * this.rowHeight}px`;
 
-                    // Insert
-                    if (rowPos >= trs.length) {
-                        tr.inject(this.tableBody);
-                        trs.push(tr);
-                    }
-                    else {
-                        tr.inject(trs[rowPos], "before");
-                        trs.splice(rowPos, 0, tr);
+            // remove existing children
+            this.tableBody.textContent = "";
+
+            // show extra 6 rows at top/bottom to reduce flickering
+            const extraRowCount = 6;
+            // how many rows can be shown in the visible area
+            const visibleRowCount = Math.ceil(this.dynamicTableDiv.offsetHeight / this.rowHeight) + (extraRowCount * 2);
+            // start position of visible rows, offsetted by scrollTop
+            const startRow = Math.max(Math.trunc(this.dynamicTableDiv.scrollTop / this.rowHeight) - extraRowCount, 0);
+            const endRow = Math.min(startRow + visibleRowCount, rows.length);
+            const fragment = document.createDocumentFragment();
+            for (let i = startRow; i < endRow; i++) {
+                const row = rows[i];
+                const isAltRow = (i % 2) === 0;
+                const offset = i * this.rowHeight;
+                const position = i - startRow;
+                // reuse existing elements
+                let element = this.cachedElements[position];
+                if (element) {
+                    this.updateRowElement(element, row["rowId"], isAltRow, offset);
+                }
+                else {
+                    element = this.cachedElements[position] = this.createRowElement(row, isAltRow, offset);
+                    // update context menu
+                    this.contextMenu?.addTarget(element);
+                }
+                fragment.appendChild(element);
+            }
+            this.tableBody.appendChild(fragment);
+
+            // update visible rows
+            for (let i = 0; i < this.tableBody.children.length; ++i)
+                this.updateRow(this.tableBody.children[i], true);
+
+            // refresh row height based on first row
+            setTimeout(() => {
```

Comment:
If possible, it should be controlled. I mean, it should be stopped when WebUI is logged out.

ps. take a look at existing code for examples on how we set the timer handler.

Reply:
my bad.